indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx

```diff
@@ -68,73 +79,85 @@ function SingleChoiceDropdown({
             (placesUsed[selectedChoice.id] || 0) - (existingValue[selectedChoice.id] || 0) >=
               selectedChoice.placesLimit)
         }
-        value={value[selectedChoice.id]}
-        onChange={(_evt, data) => onChange({[selectedChoice.id]: data.value})}
-        fluid
+        value={String(selectedSeats)}
+        onChange={evt => {
+          const selectedSlots = Number(evt.target.value);
+          onChange({[selectedChoice.id]: selectedSlots});
+        }}
+        aria-label={Translate.string('Select extra slots')}
+        required
+        {...comboBoxExtraProps}
       />
     );
   }
 
-  const options = choices.map(c => ({
-    key: c.id,
-    value: c.id,
-    disabled:
-      !c.isEnabled ||
-      isPaidChoiceLocked(c) ||
-      (c.placesLimit > 0 && (placesUsed[c.id] || 0) >= c.placesLimit && !existingValue[c.id]),
-    text: c.caption,
-    content: (
-      <div styleName="dropdown-text">
+  const options = choices.map(choice => ({
+    value: choice.caption,
+    label: (
+      <div styleName="dropdown-text" key={choice.id}>
         <div styleName="caption">
-          <ChoiceLabel choice={c} management={management} paid={isPaidChoice(c)} />
+          <ChoiceLabel choice={choice} management={management} paid={isPaidChoice(choice)} />
         </div>
         <div styleName="labels">
-          {!!c.price && (
+          {!!choice.price && (
             <Label>
-              {c.price} {currency}
+              {choice.price} {currency}
             </Label>
           )}
-          {c.placesLimit === 0 ? null : (
+          {choice.placesLimit === 0 ? null : (
             <PlacesLeft
-              placesLimit={c.placesLimit}
-              placesUsed={placesUsed[c.id] || 0}
-              isEnabled={c.isEnabled}
+              placesLimit={choice.placesLimit}
+              placesUsed={placesUsed[choice.id] || 0}
+              isEnabled={choice.isEnabled}
             />
           )}
         </div>
       </div>
     ),
+    disabled:
+      !choice.isEnabled ||
+      isPaidChoiceLocked(choice) ||
+      (choice.placesLimit > 0 &&
+        (placesUsed[choice.id] || 0) >= choice.placesLimit &&
+        !existingValue[choice.id]),
   }));
 
+  const handleChange = evt => {
+    if (!evt.target.value) {
```

Comment:
There's a bug when the field is not required and you type some nonsense in it: In that case there's no validation error, because it'll make no difference here if you explicitly cleared the field or wrote nonsense in it.

I think we need to add something similar here like what I did in the `FinalDatePicker`...

Reply:
Tbh, this is starting to sound like you don't want a combobox at all, but a searchable select list. Combobox really should be validated like any text field and then it should work as expected.

Reply:
Yeah, this was never meant to be a combobox... (more details in my [other comment](https://github.com/indico/indico/pull/6310#discussion_r1646500737))